File: SistemaGestionTransporte/src/gestorAplicaciones/camion/CamionFrigorifico.java
package gestorAplicaciones.camion;

import java.util.ArrayList;

public class CamionFrigorifico extends Camion{
    public final static ArrayList<CamionFrigorifico> camiones = new ArrayList<CamionFrigorifico>();
    public CamionFrigorifico(String placa, String pais, double pesoMaximo, double capacidad) {
        super(placa, pais, pesoMaximo, capacidad);
    }

    public static ArrayList<CamionFrigorifico> getCamiones(){
        return CamionFrigorifico.camiones;
    }

    @Override
    public double calcularCostoCamion() {
        return 0;
    }

    @Override
    public boolean comprobarPlaca(String placa) {
        return false;
    }

    @Override
    public String ubicacion() {
        return null;
    }

    @Override
    public String tiempoRestante() {
        return null;
    }

    public boolean elegirCamion(String origen, double peso, double volumen) {
        if(this.getCiudadActual().equals(origen) && peso <= this.getPesoMaximo() && volumen <= this.getCapacidad()){
            return true;
        }
        return false;
    }
}

File: python/SistemaGestionTransporte/src/gestorAplicaciones/Camion/CamionFrigorifico.py
from .camion import Camion
import pickle

class CamionFrigorifico(Camion):
    camiones = []
    
    def __init__(self, placa, pais, CiudadActual, pesoMaximo, capacidad):
        super().__init__(placa, pais, CiudadActual, pesoMaximo, capacidad)
        self.camiones.append(self)
    
    @classmethod
    def getCamiones(cls):
        return cls.camiones
    
    @classmethod
    def setCamiones(cls, camiones):
        cls.camiones = camiones
    
    def calcularCostoCamion(self):
        factor = 0.01
        km = self.getRuta()[-1].getValue()
        self.setCosto(km * self.getCapacidad() * factor)
    
    def valocidad(self):
        velocidadBase = 85.14
        factor = -0.357
        return velocidadBase + self.getCapacidad() * factor
    
    def __str__(self):
        return "\nTipo: Frigorifico" \
               "\nPlaca: " + self.getPlaca() \
               + "\npais: " + self.getPais() \
               + "\nCiudad actual: " + self.getCiudadActual() \
               + "\nPeso Maximo: " + str(self.getPesoMaximo()) \
               + "\nCapacidad: " + str(self.getCapacidad()) \
               + "\nDisponible: " + str(self.isDisponible()) + "\n"